textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/queryrewriter/QueryRewriter.java

```diff
@@ -61,18 +63,29 @@ public void open() throws Exception {
      */
     @Override
     public ITuple getNextTuple() throws Exception {
-        List<String> queryStrings = Arrays.asList(searchQuery);
-        IField[] iFieldResult = {new ListField(queryStrings)};
-        itupleResult = new DataTuple(SCHEMA_QUERY_LIST, iFieldResult);
-        return itupleResult;
+
+        boolean NOT_OPEN = (fuzzyTokenizer == null);    //Ensures QueryRewriter is open before calling getNextTuple
+        boolean EOF = (itupleResult != null);   //Ensures you can call QueryRewriter.getNextTuple only once
+
+        if(NOT_OPEN || EOF)
+            return null;
+        else {
+            List<String> queryStrings = fuzzyTokenizer.getFuzzyTokens();
+            IField[] iFieldResult = {new ListField(queryStrings)};
+            itupleResult = new DataTuple(SCHEMA_QUERY_LIST, iFieldResult);
+            return itupleResult;
+        }
     }
 
     /**
      * Closing the query Parser object
      * @throws Exception
      */
+    //TODO - take care of the case where close is called before open
```

Comment:
@chenlica TODO is implicitly handled, and has been removed in 23c809d73292b986388cb581f5d3ada0428bf077